### ISSUE TITLE
[Feature Request] Enhance UX by Storing Database Connection States in Memory

### ISSUE THREAD

**Issue by iamfj** [cweijan/vscode-mysql#1081]
**Problem Description:**

Our team uses the Database Client Extension extensively in our VSCode Team Project Workspaces. We have observed a recurrent issue where every time we close and open database connections, it alters the `database-client-config.json` file located in the `.vscode` directory. Specifically, it changes the "disabled" property to either true or false.

```json
{
    "database": {
        "1710523471234": {
            "key": "1710523471234",
            "host": "postgres",
            "port": 5432,
            (...),
            "connectionKey": "database.connections",
            "disable": true # When a connection is closed
        },
    }
}
```

```json
{
    "database": {
        "1710523471234": {
            "key": "1710523471234",
            "host": "postgres",
            "port": 5432,
            (...),
            "connectionKey": "database.connections",
            "disable": false # When a connection is opened
        },
    }
}
```

This behavior becomes quite problematic in daily operations, especially as we frequently open connections to different databases during our development cycle. The constant modification of the workspace configuration file by changing the "disabled" status of database connections disrupts our workflow and creates unnecessary version control changes, which complicates the collaborative development process.

**Proposed Solution:**

We propose that the "disabled" state of database connections should not be stored in the workspace configuration files (`database-client-config.json`) but instead be managed in local memory. This approach would ensure that each user can maintain their own state independently, without affecting the shared workspace configuration. This change would significantly improve usability and reduce the overhead of handling unnecessary changes in version control systems.

**Benefits:**

1. **User-Specific Connection States**: Each developer can manage their connection states independently without impacting others or the shared project configuration.
2. **Reduced Version Control Noise**: Keeps the `.vscode` directory cleaner and reduces unnecessary changes, making version control more about significant changes rather than configuration toggles.
3. **Enhanced Workflow Efficiency**: Developers can switch between database connections seamlessly without the hassle of constantly updating and committing workspace configuration changes.

**Conclusion:**

Implementing this feature would streamline our development process by allowing individual workspace environments to remain static while still enabling personal connection management. This enhancement aligns with the general principle of separating personal workspace settings from shared project settings, thus fostering a more efficient and less disruptive development environment.

Thank you for considering this feature request. We believe it will significantly benefit many users of the Database Client Extension by providing a more personalized and less intrusive user experience.

**Comment by cweijan** [comment]
Thanks for the feedback, it’s a great suggestion. I will support this feature in the next version.

**Comment by iamfj** [comment]
Thank you for your prompt reply! I'm eagerly anticipating the next version—your dedication to crafting such an outstanding extension is truly appreciated! 🚀

**Comment by cweijan** [comment]
Implemented in version 7.1.2.

**Comment by iamfj** [comment]
@cweijan, the latest version has been shared with the team. However, we've noticed an issue where the disabled state persists in the file upon opening and closing connections. This results in changes appearing in the git view.

<img width="732" alt="image" src="https://github.com/cweijan/vscode-database-client/assets/24557998/2d0dc835-d5dd-4b41-9c67-5b6918563cff">
